### ISSUE TITLE
Enforce REST API constraints

### ISSUE THREAD

**Issue by shekhargulati** [xebia-os/xebia-archunit-extension#5]
- [ ] Resources should be noun

- [ ] Use kebab-case instead of camelCase for API URLs

- [ ] Plural for resource names

- [ ] Limit use of subresource

- [ ] Don’t use a trailing forward slash

- [ ] Use ResponseEntity

- [ ] Check response status code

**Comment by shekhargulati** [comment]
@awkshwayrd do you want to pick it up?

**Comment by axymthr** [comment]
@shekhargulati Should we split up the rules classes? Right now there is just one. It could be segregated by layers.

**Comment by shekhargulati** [comment]
Yes we can do that. I think it is a good idea. Should we have a rule interface that all rules should implement? We can then chain them to build the rule pipeline.

**Comment by axymthr** [comment]
Would it take us away from the basic ArchRule interface? Right now we have 1 single class defining static rules and a run-all-tests abstract test class. This allows clients to just use our defined rules. If we put in more layers we would be hiding the basic ArchUnit abstration.

**Comment by axymthr** [comment]
We could split up AbstractArchitectureTests as well, but it starts to go down YAGNI a little bit.
